optuna/samplers/_grid.py

```diff
@@ -89,6 +89,11 @@ def objective(trial):
         suggested may vary. This is to reduce duplicate suggestions during distributed
         optimization.
 
+    Note:
+        When the trial is created by :meth:`~optuna.study.Study.enqueue_trial`,
+        :class:`~optuna.samplers.GridSampler` does not consider the trial to be a grid. We should
+        specify all parameters with :meth:`~optuna.study.Study.enqueue_trial` in this case.
```

Comment:
```suggestion
        All parameters must be specified when using :class:`~optuna.samplers.GridSampler` with :meth:`~optuna.study.Study.enqueue_trial`.
```
Not sure it's necessary here to talk about the internal mechanics of whether Gridsampler considers the trial a grid or not. Just let the user know they have to specify all arguments.